bytestring/common/src/Hex.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+@file:OptIn(ExperimentalStdlibApi::class)
+
+package kotlinx.io.bytestring
+
+/**
+ * Formats bytes in this byte string using the specified [format].
+ *
+ * Note that only [HexFormat.upperCase] and [HexFormat.BytesHexFormat] affect formatting.
+ *
+ * @param format the [HexFormat] to use for formatting, [HexFormat.Default] by default.
+ *
+ * @throws IllegalArgumentException if the result length is more than [String] maximum capacity.
+ */
+@ExperimentalStdlibApi
+@SinceKotlin("1.9")
```

Comment:
I'm not sure if this is applicable anywhere outside of the standard library. Ilya may know for sure